src/inference/src/os/win/win_system_conf.cpp

```diff
@@ -7,21 +7,139 @@
 #endif
 
 #include <windows.h>
+
+#include <iostream>
 #include <memory>
 #include <vector>
+
 #include "ie_system_conf.h"
 #include "threading/ie_parallel_custom_arena.hpp"
 
 namespace InferenceEngine {
+
 struct CPU {
     int _processors = 0;
     int _sockets = 0;
     int _cores = 0;
 
     std::vector<int> _proc_type_table;
     std::vector<std::vector<int>> _cpu_mapping_table;
+
+    CPU() {
+        DWORD len = 0;
+        if (GetLogicalProcessorInformationEx(RelationAll, nullptr, &len) ||
+            GetLastError() != ERROR_INSUFFICIENT_BUFFER) {
+            return;
+        }
+
+        char* base_ptr = new char[len];
+        if (!GetLogicalProcessorInformationEx(RelationAll, (PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX)base_ptr, &len)) {
+            return;
```

Comment:
memory leak here since `base_ptr` is not deleted. Why don't you use smart pointers?

Reply:
updated